scripts/skills/actives/rf_hook_shield_skill.nut

```diff
@@ -110,20 +113,38 @@ this.rf_hook_shield_skill <- ::inherit("scripts/skills/skill", {
 		if (targetEntity.isAlliedWith(this.getContainer().getActor()))
 			return false;
 
-		return targetEntity.isArmedWithShield() && this.skill.onVerifyTarget(_originTile, _targetTile) && !targetEntity.getSkills().hasSkill("effects.shieldwall") && !targetEntity.getSkills().hasSkill("effects.rf_hooked_shield");
+		return targetEntity.isArmedWithShield() && this.skill.onVerifyTarget(_originTile, _targetTile) && !targetEntity.getSkills().hasSkill("effects.rf_hooked_shield");
 	}
 
 	function onUse( _user, _targetTile )
 	{
 		local targetEntity = _targetTile.getEntity();
+
+		if (!this.m.IsSpent && targetEntity.getID() == this.m.LastTargetID)
+		{
+			_user.setActionPoints(::Math.min(_user.getActionPointsMax(), _user.getActionPoints() + this.getActionPointCost()));
+			this.m.IsSpent = true;
+		}
+
+		this.m.LastTargetID = targetEntity.getID();
+
 		local shield = targetEntity.getOffhandItem();
 		if (shield != null)
 		{
 			this.spawnAttackEffect(_targetTile, ::Const.Tactical.AttackEffectSplitShield);
 			if (this.attackEntity(_user, targetEntity))
 			{
-				::Tactical.EventLog.log(::Const.UI.getColorizedEntityName(_user) + " hooks " + ::Const.UI.getColorizedEntityName(targetEntity) + "\'s shield");
-				targetEntity.getSkills().add(::new("scripts/skills/effects/rf_hooked_shield_effect"));
+				local shieldWall = targetEntity.getSkills().getSkillByID("effects.shieldwall");
+				if (shieldWall != null)
+				{
+					::Tactical.EventLog.log(::Const.UI.getColorizedEntityName(_user) + " hooks " + ::Const.UI.getColorizedEntityName(targetEntity) + "\'s shield removing their " + shieldWall.m.Name);
+					targetEntity.getSkills().removeByID("effects.shieldWall");
```

Comment:
You could use `(shieldWall.getID())` here or `.remove(shieldWall)`, to tie it closer to the already fetched shieldwall skill from before. Just in case we change it or allow more different shieldwall like effects to be the target